app/models/assignment_repo/creator.rb

```diff
@@ -189,5 +190,69 @@ def suffixed_repo_name(repository_name, suffix_count)
       suffix = "-#{suffix_count}"
       repository_name.truncate(100 - suffix.length, omission: '') + suffix
     end
+
+    # Internal: Handles assignment repository setup
+    #
+    # configs_repo    - GitHubRepository containing the configuration files
+    # assignment_repo - GitHubRepository for which to perform the configuration
+    #                   setups
+    #
+    # Returns nothing
+    def setup_repository(configs_repo, assignment_repo)
+      return unless configs_repo.branch_present? 'github-classroom'
+      configs_tree = configs_repo.branch_tree('github-classroom')
+      configs_tree.tree.each do |configuration|
+        process_configuration(configs_repo, assignment_repo, configuration)
+      end
+
+      wait_import_completion assignment_repo
+
+      assignment_repo.remove_branch('github-classroom')
+    end
+
+    # Internal: Wait for import completion
+    #
+    # github_repository - GitHubRepository awaiting import completion
+    #
+    # Returns nothing
+    def wait_import_completion(github_repository)
+      wait_limit = 60
+      increment_rate = 0.25
+      t = 1
+      loop do
+        break if github_repository.import_progress[:status] == 'complete' || t > wait_limit
+        sleep t += (t *= increment_rate).ceil
+      end
+    end
+
+    # Internal: Process the configurations separetly
+    #
+    # configs_repo    - GitHubRepository containing the configuration files
+    # assignment_repo - GitHubRepository for which to perform the configuration
+    #                   setups
+    # configuration   - GitObjects representing the configuration to process
+    #
+    # Returns nothing
+    def process_configuration(configs_repo, assignment_repo, configuration)
+      case configuration.path
+      when 'issues'
+        generate_issues(configs_repo, assignment_repo, configuration.sha)
+      end
+    end
+
+    # Internal: Generates issues for the assignment_repo based on the configs
+    #
+    # configs_repo    - GitHubRepository containing the configuration files
+    # assignment_repo - GitHubRepository for which to perform the configuration
+    #                   setups
+    # tree_sha     - sha of the 'issues' tree
+    #
+    # Returns nothing
+    def generate_issues(configs_repo, assignment_repo, tree_sha)
```

Comment:
We need some error handling here. `add_issue` raises a GitHub::Error if the API call fails which will bubble all the way up to our ApplicationController and 500 I believe.